streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -199,46 +201,18 @@
+     * @param consumed           the instance of {@link Consumed} used to define optional parameters; cannot be {@code null}
+     * @param materialized       the instance of {@link Materialized} used to materialize a state store; cannot be {@code null}
      * @return a {@link KTable} for the specified topic
      */
     public synchronized <K, V> KTable<K, V> table(final String topic,
-                                                  final String queryableStoreName) {
-        return internalStreamsBuilder.table(topic, new ConsumedInternal<K, V>(), queryableStoreName);
-    }
-
-    /**
-     * Create a {@link KTable} for the specified topic.
-     * The default {@code "auto.offset.reset"} strategy and default key and value deserializers as specified in the
-     * {@link StreamsConfig config} are used.
-     * Input {@link KeyValue records} with {@code null} key will be dropped.
-     * <p>
-     * Note that the specified input topics must be partitioned by key.
-     * If this is not the case the returned {@link KTable} will be corrupted.
-     * <p>
-     * The resulting {@link KTable} will be materialized in a local {@link KeyValueStore} with the given
-     * {@code queryableStoreName}.
-     * However, no internal changelog topic is created since the original input topic can be used for recovery (cf.
-     * methods of {@link KGroupedStream} and {@link KGroupedTable} that return a {@link KTable}).
-     * <p>
-     * To query the local {@link KeyValueStore} it must be obtained via
-     * {@link KafkaStreams#store(String, QueryableStoreType) KafkaStreams#store(...)}:
-     * <pre>{@code
-     * KafkaStreams streams = ...
-     * ReadOnlyKeyValueStore<String, Long> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, Long>keyValueStore());
-     * String key = "some-key";
-     * Long valueForKey = localStore.get(key); // key must be local (application state is shared over all running Kafka Streams instances)
-     * }</pre>
-     * For non-local keys, a custom RPC mechanism must be implemented using {@link KafkaStreams#allMetadata()} to
-     * query the value of the key on a parallel running instance of your Kafka Streams application.
-     *
-     * @param topic         the topic name; cannot be {@code null}
-     * @param storeSupplier user defined state store supplier; cannot be {@code null}
-     * @return a {@link KTable} for the specified topic
-     */
-    public synchronized <K, V> KTable<K, V> table(final String topic,
-                                                  final StateStoreSupplier<KeyValueStore> storeSupplier) {
-        return internalStreamsBuilder.table(null, null, null, null, topic, storeSupplier);
+                                                  final Consumed<K, V> consumed,
+                                                  final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(consumed, "consumed can't be null");
```

Comment:
This is not introduced in this PR: seems we do not check topic name is `null` anywhere for both `stream` and `table/globalTable`, is this intentional?

Reply:
Also a quick question: if `Consumed` does not specify the same serde as `Materialized`, should we just use different serdes then? I'm asking this mainly because today we will do a deser reading from Kafka and then a ser writing to state store, and maybe we can avoid this deser/ser together as an optimization. But if we allow different serdes here we cannot do that.

Reply:
I think we should force them to be the same, see comment above. I'll update